machine/v1beta1/types_azureprovider.go

```diff
@@ -215,6 +223,48 @@ type OSDisk struct {
 	CachingType string `json:"cachingType,omitempty"`
 }
 
+// DataDisk specifies the parameters that are used to add one or more data disks to the machine.
+type DataDisk struct {
+	// NameSuffix is the suffix to be appended to the machine name to generate the disk name.
+	// Each disk name will be in format <machineName>_<nameSuffix>.
+	// NameSuffix name must start and finish with an alphanumeric character and can only contain letters, numbers, underscores, periods or hyphens.
+	// The overall disk name must not exceed 80 chars in length.
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9](?:[\w\.-]*[a-zA-Z0-9])?$`
```

Comment:
Note: I had to change the regex a little from what was described in the enhancement (which I will update once this PR has been reviewed).
I had to use backticks rather than double quotes to wrap the regex, to make controller-gen happy (would've had to escape `\` otherwise). I've also pinned beginning and end of the string with `^` and `$`.